files/static/index.js

```diff
@@ -0,0 +1,340 @@
+      oReq.addEventListener("load", (function(ev) {
+        resolve(JSON.parse(oReq.response));
+      }).bind(this));
+      oReq.open("GET", "/checkdomain/" + domain);
+      oReq.send();
+    });
+  },
+
+  submit: function(domain) {
+    console.log("checkdomain:" + domain);
+    return new Promise(function(resolve, reject) {
+      var oReq = new XMLHttpRequest();
+      oReq.addEventListener("load", (function(ev) {
+        resolve(JSON.parse(oReq.response));
+      }).bind(this));
+      oReq.open("GET", "/submit/" + domain);
+      oReq.send();
+    });
+  },
+
+  isValidDomainFormat: function(domain) {
+    return domain.includes(".") &&
+          !domain.endsWith(".") &&
+          !domain.startsWith(".") &&
+          !domain.includes("..");
+  },
+
+  // hasLikelyETLD1: function(domain) {
+  //   // Submitting .com domains is common; don't use the fast path for .co
+  //   // Same for .net => .ne
+  //   if (domain.endsWith(".co") || domain.endsWith(".ne")) {
+  //     return true;
+  //   }
+
+  //   var publicsuffix = tldjs.getPublicSuffix(domain)
+  //   return publicsuffix !== null && publicsuffix !== domain;
+  // },
+
+  clearTheme: function(theme) {
+    document.body.classList.remove("theme-green", "theme-yellow", "theme-red");
+  },
+
+  setTheme: function(theme) {
+    this.clearTheme();
+    document.body.classList.add(theme)
+  },
+
+  currentDomain: function() {
+    var domain = $("#domain").value;
+    // Check for pasted URLs beginning with http:// or https://
+    if (domain.startsWith("http://")) {
+      domain = domain.slice("http://".length)
+    }
+    if (domain.startsWith("https://")) {
+      domain = domain.slice("https://".length)
+    }
+    return domain;
+  },
+
+  clearResults: function() {
+    $("#result").classList.add("hidden");
+  },
+
+  showResults: function(domain, checkDomainIssues, domainStatus) {
+    if (checkDomainIssues.errors.length === 0) {
+      if (checkDomainIssues.warnings.length === 0) {
+        $("#summary").textContent = "" + domain + " satisfies all requirements for preloading!";
+        this.setTheme("theme-green");
+      } else {
+        $("#summary").textContent = "" + domain + " has " + (checkDomainIssues.warnings.length == 1 ? "a warning" : "warnings") + ", but satisfies all requirements for preloading.";
+        this.setTheme("theme-yellow");
+      }
+    } else {
+        $("#summary").textContent = "" + domain + " has errors and cannot be preloaded.";
+        this.setTheme("theme-red");
+    }
+
+    switch (domainStatus.status) {
+    case "unknown":
+      if (checkDomainIssues.errors.length === 0) {
+        $("#status").textContent = "" + domain + " is not yet preloaded, and may be submitted to the preload list."
+      } else {
+        $("#status").textContent = "" + domain + " is not preloaded."
+      }
+      break;
+    case "pending":
+      $("#status").textContent = "" + domain + " is pending submission to the preload list."
```

Comment:
In here and elsewhere, prefer single quotes over doubles: https://google.github.io/styleguide/javascriptguide.xml?showone=Strings#Strings